gateway/api_definition.go

```diff
@@ -605,7 +601,6 @@ func (a APIDefinitionLoader) processRPCDefinitions(apiCollection string, gw *Gat
 	// Extract tagged entries only
 	apiDefs := list.filter(gwConfig.DBAppConfOptions.NodeIsSegmented, gwConfig.DBAppConfOptions.Tags...)
 
-	var specs []*APISpec
 	for _, def := range apiDefs {
 		def.DecodeFromDB()
```

Comment:
The modifications to `def` in this context... would be nice to move to prepareSpecs, and remove this loop. Currently it raises a code smell, as it should not modify nestedApiDefinition values (non-pointers).